principal/server.go

```diff
@@ -274,6 +277,37 @@ func (s *Server) Start(ctx context.Context, errch chan error) error {
 	return nil
 }
 
+// removeUnusedQueues removes the queues that are not associated with any namespace/agent.
+func (s *Server) removeUnusedQueues(ctx context.Context) {
+	ticker := time.NewTimer(10 * time.Minute)
+	for {
+		select {
+		case <-s.ctx.Done():
+			return
+		case <-ticker.C:
+			queues := s.queues.Names()
+			for _, queueName := range queues {
+				s.removeQueueIfUnused(ctx, queueName)
+			}
+		}
+	}
+}
+
+func (s *Server) removeQueueIfUnused(ctx context.Context, agentName string) {
```

Comment:
I think the better solution would be to remove the queue when the agent (in case of Kubernetes backend: namespace) is deleted., instead of executing on an interval.

Reply:
Updated. I have included a simple namespace informer using the generic informer in the backend module.